code/modules/projectiles/guns/flamer.dm

```diff
@@ -447,6 +437,15 @@
 
 	new /obj/flamer_fire(src, fire_lvl, burn_lvl, f_color, fire_stacks, fire_damage)
 
+	for(var/obj/structure/jungle/vines/vines in src)
+		QDEL_NULL(vines)
+
+/turf/open/floor/plating/ground/snow/ignite(fire_lvl, burn_lvl, f_color, fire_stacks = 0, fire_damage = 0)
+	if(slayer > 0)
+		slayer -= 1
+		update_icon(1, 0)
+	return ..()
+
```

Comment:
This does not affect only flamers, but anything that uses ignite(), which is a bunch of other things which are not flamers. So, less a fix and more a balance change given how much this touches.

Reply:
You're right and I overlooked that, but I'm also completely fine with fire sources generally killing vines and snow. I'll toss it in the changelog.